components/Currency/CurrencyIcon/CurrencyIcon.tsx

```diff
@@ -116,206 +15,50 @@ type CurrencyIconProps = {
 	height?: string;
 };
 
-export const CurrencyIcon: FC<CurrencyIconProps> = ({ currencyKey, type = 'synth', ...rest }) => {
+export const SNXIcon =
+	'https://raw.githubusercontent.com/Synthetixio/synthetix-assets/master/snx/SNX.svg';
+
+export const getSynthIcon = (currencyKey: CurrencyKey) =>
+	`https://raw.githubusercontent.com/Synthetixio/synthetix-assets/master/synths/${currencyKey}.svg`;
+
+export const CurrencyIcon: FC<CurrencyIconProps> = ({ currencyKey, ...rest }) => {
+	const [isError, setIsError] = useState<boolean>(false);
+
 	const props = {
 		width: '24px',
 		height: '24px',
 		alt: currencyKey,
 		...rest,
 	};
 
-	// TODO: next-optimized-images does not support dynamic imports yet... so it needs to be manually defined.
+	if (isError) {
+		return (
+			<Placeholder style={{ width: props.width, height: props.height }}>{currencyKey}</Placeholder>
+		);
+	}
 
-	// most of the "asset" types were disabled since they were not widely used.
 	switch (currencyKey) {
 		case CRYPTO_CURRENCY_MAP.ETH: {
 			return <Img src={ETHIcon} {...props} />;
```

Comment:
I left ETH icon intentionally here to be loaded from Kwenta because its great quality... Once a good alternative is found, will switch to that. (a high-res place to load crypto icons).